src/lib/mixer_module/mixer_module.cpp

```diff
@@ -322,6 +321,28 @@ bool MixingOutput::update()
 		return false;
 	}
 
+	if (_vehicle_land_detected_sub.updated()) {
+		vehicle_land_detected_s vehicle_land_detected;
+
+		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
+
+			if (vehicle_land_detected.landed || vehicle_land_detected.maybe_landed) {
+				// disable airmode when landed to settle on uneven ground
+				_mixers->set_airmode(Mixer::Airmode::disabled);
+
+			} else {
+				const Mixer::Airmode airmode = static_cast<Mixer::Airmode>(_param_mc_airmode.get());
```

Comment:
This is all correct and better than before thanks. Wouldn't it generally make sense to do the sanity checks inside the setter `set_airmode()` and just either fill in the parameter or set it to disabled here?

Reply:
Yes pushing it all the way in would be the cleanest thing unless we want to actually change the `MC_AIRMODE` parameter value for some reason (reset bogus value, maybe disable airmode on a bad setup, etc).